src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -792,13 +792,14 @@ bool CanRemove(Hash256? address, TinyTreePath path, in TreePath fullPath, ValueH
             void DoAct(KeyValuePair<HashAndTinyPath, Hash256> keyValuePair)
             {
                 HashAndTinyPath key = keyValuePair.Key;
-                if (_pastPathHash.TryGet(new(key.addr, in key.path), out ValueHash256 prevHash))
+                if (_pastPathHash.TryGet(key, out ValueHash256 prevHash))
                 {
                     TreePath fullPath = key.path.ToTreePath(); // Micro op to reduce double convert
                     if (CanRemove(key.addr, key.path, fullPath, prevHash, keyValuePair.Value))
                     {
                         Metrics.RemovedNodeCount++;
                         writeBatch.Remove(key.addr, fullPath, prevHash);
+                        _pastPathHash.Delete(key);
```

Comment:
Will this make mem pruning slower, (since it does more work), or faster? (since it become smaller).